packages/local_auth/local_auth_ios/lib/local_auth_ios.dart

```diff
@@ -2,85 +2,119 @@
+    // The PlatformExceptions thrown here are for compatibiilty with the
+    // previous Objective-C implementation.
+    switch (result.value) {
+      case AuthResult.success:
+        return true;
+      case AuthResult.failure:
+        return false;
+      case AuthResult.errorNotAvailable:
+        throw PlatformException(
+            code: 'NotAvailable',
+            message: result.errorMessage,
+            details: result.errorDetails);
+      case AuthResult.errorNotEnrolled:
+        throw PlatformException(
+            code: 'NotEnrolled',
+            message: result.errorMessage,
+            details: result.errorDetails);
+      case AuthResult.errorPasscodeNotSet:
+        throw PlatformException(
+            code: 'PasscodeNotSet',
+            message: result.errorMessage,
+            details: result.errorDetails);
     }
-    return (await _channel.invokeMethod<bool>('authenticate', args)) ?? false;
   }
 
   @override
   Future<bool> deviceSupportsBiometrics() async {
-    return (await _channel.invokeMethod<bool>('deviceSupportsBiometrics')) ??
-        false;
+    return _api.deviceCanSupportBiometrics();
   }
 
   @override
   Future<List<BiometricType>> getEnrolledBiometrics() async {
-    final List<String> result = (await _channel.invokeListMethod<String>(
-          'getEnrolledBiometrics',
-        )) ??
-        <String>[];
-    final List<BiometricType> biometrics = <BiometricType>[];
-    for (final String value in result) {
-      switch (value) {
-        case 'face':
-          biometrics.add(BiometricType.face);
-          break;
-        case 'fingerprint':
-          biometrics.add(BiometricType.fingerprint);
-          break;
-        case 'iris':
-          biometrics.add(BiometricType.iris);
-          break;
+    final List<AuthBiometricWrapper?> result =
+        await _api.getEnrolledBiometrics();
+    return result
+        .cast<AuthBiometricWrapper>()
+        .map((AuthBiometricWrapper entry) {
+      switch (entry.value) {
+        case AuthBiometric.face:
+          return BiometricType.face;
+        case AuthBiometric.fingerprint:
+          return BiometricType.fingerprint;
       }
-    }
-    return biometrics;
+    }).toList();
   }
 
   @override
-  Future<bool> isDeviceSupported() async =>
-      (await _channel.invokeMethod<bool>('isDeviceSupported')) ?? false;
+  Future<bool> isDeviceSupported() async => _api.isDeviceSupported();
 
   /// Always returns false as this method is not supported on iOS.
   @override
-  Future<bool> stopAuthentication() async {
-    return false;
+  Future<bool> stopAuthentication() async => false;
+
+  AuthStrings _pigeonStringsFromAuthMessages(
+      String localizedReason, Iterable<AuthMessages> messagesList) {
+    IOSAuthMessages? messages;
+    for (final AuthMessages entry in messagesList) {
+      if (entry is IOSAuthMessages) {
+        messages = entry;
```

Comment:
can probably do `firstWhere`

Reply:
It's tricky; the original logic (line 40+ in the old file) would, in the unlikely event of there being multiple entries for the same platform, layer them all together; this wasn't deliberate behavior, but it has been what would happen since the federation changes.

I opted for taking the last as being the closest to the existing behavior, but since it's going to change slightly either way, and it would only affect people who we doing something very bizarre (hopefully nobody) and undocumented, taking the first probably makes sense. I've added `break`.